### PR TITLE
doc: update files to main for NCS 2.4.2

### PR DESCRIPTION
**Pull request by peknis**
Updating known issues and adding release notes to the main branch.

### CODE REVIEW

File: doc/nrf/releases_and_maturity/known_issues.rst [hunk 1]
```diff
@@ -1014,7 +1015,7 @@ KRKNWK-14130: Bluetooth LE TX configuration is set to ``0`` dBm by default
 KRKNWK-14081: HomeKit SDK light bulb example does not work with MTD
   If the MTD is set to ``y`` in the light bulb sample, user is not able to communicate with the device after it has been added to the Home app using an iPhone and a HomePod Mini.
 
-.. rst-class:: v2-4-1 v2-4-0 v2-3-0 v2-2-0 v2-1-4 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0
+.. rst-class:: v2-4-2 v2-4-1 v2-4-0 v2-3-0 v2-2-0 v2-1-4 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0
```
Comment (mariuszpos) [inline doc/nrf/releases_and_maturity/known_issues.rst:1018]: Please remove v2-4-2 tag due to fix: https://github.com/nrfconnect/sdk-nrf/pull/12095

**Review by mariuszpos — APPROVED** [review]

**Comment by umapraseeda** [comment]
Shouldn't NCSDK-13947 have 2.4.2 tag?

**Review by umapraseeda — CHANGES_REQUESTED** [review]
Update missing for shortcuts.txt

**Comment by umapraseeda** [comment]
> Update missing for shortcuts.txt

Also for dm_adding_code.rst.

**Comment by peknis** [comment]
> NCSDK-13947

No, this was fixed in 2.4.2.

**Review by umapraseeda — APPROVED** [review]

**Review by b-gent — APPROVED** [review]

**Review by nordic-hani — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**